packages/swapper/src/utils.ts

```diff
@@ -179,14 +180,32 @@ export const checkEvmSwapStatus = async ({
   message: string | undefined
 }> => {
   try {
+    const safeTransactionInfo = await fetchSafeTransactionInfo({ chainId, maybeSafeTxHash: txHash })
+    const { isSafeTxHash, transaction } = safeTransactionInfo
+
+    // No buyTxHash handling is correct - we mutate with the actual on-chain transaction, meaning the regular flow then takes over
+    if (
+      isSafeTxHash &&
+      transaction?.confirmations &&
+      Number(transaction.confirmations.length) < transaction.confirmationsRequired
+    ) {
+      return {
+        status: TxStatus.Pending,
+        message: `SAFE proposal submitted. ${transaction.confirmations.length} out of ${transaction.confirmationsRequired} signed.`,
+        buyTxHash: undefined,
+      }
+    } else if (transaction?.transactionHash) {
+      // Mutate with  the actual on-chain transaction work and let things work as-is
+      txHash = transaction.transactionHash
+    }
     const adapter = assertGetEvmChainAdapter(chainId)
     const tx = await adapter.httpProvider.getTransaction({ txid: txHash })
     const status = getTxStatus(tx)
 
     return {
       status,
       buyTxHash: txHash,
-      message: undefined,
+      message: transaction?.transactionHash ? `SAFE proposal executed.` : undefined,
```

Comment:
Thorchain not leveraging the same confirm message btw

Reply:
discussed in dms this is intentional for thorchain inbound vernacular.

Reply:
And actually going away with the refactor! Leveraging safe-first terminology for the sake of simplicity.